app/frontend/src/pages/chat/Chat.tsx

```diff
@@ -183,16 +183,20 @@ const Chat = () => {
             };
 
             const response = await chatApi(request, shouldStream, token);
+
             if (!response.body) {
                 throw Error("No response body");
             }
+            if (response.status > 299 || !response.ok) {
```

Comment:
Same change here, throw a non-200 error earlier before looking for error in the JSON